samurai.c

```diff
@@ -95,6 +97,12 @@ main(int argc, char *argv[])
 	argv0 = strrchr(argv[0], '/');
 	argv0 = argv0 ? argv0 + 1 : argv[0];
 	ARGBEGIN {
+	case '-':
+		if (strcmp(&argv[0][1], "version") == 0) {
+			puts(NINJA_VERSION);
+			exit(0);
+		}
```

Comment:
I think you should call `usage()` here if the long option is not known.

Reply:
done